cpp/src/io/json/write_json.cu

```diff
@@ -86,6 +86,52 @@ struct escape_strings_fn {
       bytes += cudf::strings::detail::bytes_in_char_utf8(chr);
   }
 
+  __device__ inline char nibble_to_hex(uint8_t nibble) const
+  {
+    return nibble < 10 ? '0' + nibble : 'a' + nibble - 10;
+  }
+
+  __device__ void write_utf8_codepoint(uint32_t codepoint, char*& d_buffer, offset_type& bytes)
+  {
+    if (d_buffer) {
+      d_buffer[0] = '\\';
+      d_buffer[1] = 'u';
+      d_buffer[2] = nibble_to_hex((codepoint >> 12) & 0x0F);
+      d_buffer[3] = nibble_to_hex((codepoint >> 8) & 0x0F);
+      d_buffer[4] = nibble_to_hex((codepoint >> 4) & 0x0F);
+      d_buffer[5] = nibble_to_hex((codepoint)&0x0F);
```

Comment:
Can we use intrinsics?
```cuda
  void __device__ bytes_to_hex(uint16_t codepoint, char* d_buffer)
  {
    auto nibble_in_bytes =  __byte_perm(codepoint, (codepoint >> 4), 0x0415) & 0x0f0f'0f0f;
    auto nibbles_to_ascii = [](auto bytes) {
      // int + '0' + (int > 9) * ('a' - '0' - 10)
      return bytes + 0x3030'3030 + (__vcmpgtu4(bytes, 0x0909'0909) & 0x2727'2727);
    };
    auto nibble_as_ascii = nibbles_to_ascii(nibble_in_bytes);
    memcpy(d_buffer, &nibble_as_ascii, 4);
  }
```

```suggestion
      bytes_to_hex(codepoint, d_buffer+2);
```

Reply:
It barely makes any difference. Improved only 3us on avg, 5us on median. Besides, this contributes to only 0.1% of total runtime. so, skipping this idea.
```
INTRINSICS:
 Time (%)  Total Time (ns)  Instances    Avg (ns)      Med (ns)     Min (ns)    Max (ns)   StdDev (ns)   Style                        Range                      
      0.1         11638826        110      105807.5       81410.5       68185      156432      33467.6  PushPop  libcudf:get_escaped_strings::get_escaped_strings
      0.1          6358774         40      158969.4      158641.0      150164      169667       4462.8  PushPop  libcudf:operator()::string_view                 

BEFORE:
 Time (%)  Total Time (ns)  Instances    Avg (ns)      Med (ns)     Min (ns)    Max (ns)   StdDev (ns)   Style                        Range                      
      0.2         11905546        110      108232.2       86092.0       69701      163563      33932.9  PushPop  libcudf:get_escaped_strings::get_escaped_strings
      0.1          6441591         40      161039.8      159258.0      153868      177243       5139.8  PushPop  libcudf:operator()::string_view                 
```

Reply:
@davidwendt any opinion on using intrinsics.

Reply:
Since the intrinsics are not significantly faster, I'd prefer the code be easier to read, maintain. 
Very cool though.